cli/lsp/diagnostics.rs

```diff
@@ -490,23 +501,39 @@ async fn generate_ts_diagnostics(
     .iter()
     .map(|d| d.specifier().clone())
     .collect::<Vec<_>>();
-  if !specifiers.is_empty() {
-    let req = tsc::RequestMethod::GetDiagnostics(specifiers);
-    let ts_diagnostics_map: TsDiagnosticsMap =
-      ts_server.request(snapshot.clone(), req).await?;
-    for (specifier_str, ts_diagnostics) in ts_diagnostics_map {
-      let specifier = resolve_url(&specifier_str)?;
-      let version = snapshot
-        .documents
-        .get(&specifier)
-        .map(|d| d.maybe_lsp_version())
-        .flatten();
-      diagnostics_vec.push((
-        specifier,
-        version,
-        ts_json_to_diagnostics(ts_diagnostics),
-      ));
-    }
+  let (enabled_specifiers, disabled_specifiers) = specifiers
+    .iter()
+    .cloned()
+    .partition::<Vec<_>, _>(|s| config.specifier_enabled(s));
+  let ts_diagnostics_map: TsDiagnosticsMap = if !enabled_specifiers.is_empty() {
+    let req = tsc::RequestMethod::GetDiagnostics(enabled_specifiers);
```

Comment:
I think only doing this for enabled specifiers should be a decent performance improvement in workspaces with a lot of disabled files.